ansible/pbTestScripts/testScript.sh

```diff
@@ -110,24 +167,33 @@ splitURL()
 		done
 	else
 		folderName=${array[@]: -1:1}
-		gitURL=$1
 	fi
 }
 # var1 = GitURL, var2 = y/n for VM retention
 processArgs $*
-splitURL $1
+splitURL
+checkVagrantOS
 setupFiles
 setupGit
-# For all tested OSs / Playbooks
-for OS in Ubuntu1804 Ubuntu1604 CentOS6 CentOS7
-do
-	startVMPlaybook $OS
-	if [[ $2 = "n" ]]
-	then
-		destroyVM
-	fi
-done
-for OS in Ubuntu1804 Ubuntu1604 CentOS6 CentOS7
-do
-	searchLogFiles $OS $branchName
-done
+# Testing all of the OSs
```

Comment:
To avoid duplication I'd suggest setting `vagrantOS` to `"Ubuntu1604 Ubuntu1804 CentOS6 CentOS7"` in the parameter parsing section as a default, and then usiong `for OS in $vagrantOS` - that will render this if section irrelevant.